languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java

```diff
@@ -48,33 +48,30 @@ public String getDescription() {
     return "'dhá' nó 'beirt'";
   }
 
-  private String[] getRestOfNP(AnalyzedTokenReadings[] tokens, int start, int end) {
-    List<String> ret = new ArrayList<>();
-    for (int i = start; i <= end; i++) {
-      ret.add(tokens[i].getToken());
-    }
-    return ret.toArray(new String[ret.size()]);
-  }
-
   @Override
   public RuleMatch[] match(AnalyzedSentence sentence) {
     List<RuleMatch> ruleMatches = new ArrayList<>();
     AnalyzedTokenReadings[] tokens = sentence.getTokensWithoutWhitespace();
-    int markEnd = 1;
+    int markDeag = 1;
     int prevTokenIndex = 0;
     String replacement = null;
     String msg = null;
     for (int i = 1; i < tokens.length; i++) {  // ignoring token 0, i.e., SENT_START
       if (isNumber(tokens[i]) && (i < tokens.length - 2 && isPerson(tokens[i + 1]))) {
-        markEnd = i + 1;
         if ("dhá".equalsIgnoreCase(tokens[i].getToken()) && (i < tokens.length - 2)) {
           for (int j = i + 2; j < tokens.length; j++) {
             if ("déag".equalsIgnoreCase(tokens[j].getToken())) {
-              markEnd = j;
+              markDeag = j;
               replacement = "dháréag";
-              String[] msgTokens = getRestOfNP(tokens, i + 1, j - 1);
-              msg = "Ba chóir duit <suggestion>" + replacement + " " +
-                String.join(" ", msgTokens) + "</suggestion> a scríobh";
+              msg = "Ba chóir duit <suggestion>" + replacement + "</suggestion> a scríobh";
+              RuleMatch match = new RuleMatch(
+                this, sentence, tokens[prevTokenIndex].getStartPos(), tokens[prevTokenIndex].getEndPos(), msg, "Uimhir phearsanta");
+              ruleMatches.add(match);
+              msg = "Ba chóir duit \"déag\" a scriosadh.";
+              RuleMatch match2 = new RuleMatch(
```

Comment:
One issue with adding all matches in the same conditional is that if you replace the first match, the pending word (déag) will remain in the sentence. Your previous solution did not have this problem.

Reply:
Trust me, this sucks less

Reply:
Thank you for the fix @jimregan . Now, only the fix for the replacement for single numerals is missing. It is duplicating the noun the adjacent noun. It is also not a problem when the marker covers all the related words.